### ISSUE TITLE
Spotify does not work

### ISSUE THREAD

**Issue by drbubo00** [ElToberino/Tobers_Multidisplay#3]
Hi,
Thank you for this great code. Everything works properly except spotify.
I set the client id and client secret data in the code. I did everything the way you wrote it.
The description does not include the fingerprint, however, the code does include "necessary for https connection, check here, e.g .: https://www.grc.com/fingerprints.htm"
I don't understand what to enter after https: // at on https://www.grc.com/fingerprints.htm webpage?

**Comment by victor7376** [comment]
I’ll check that today, but on first look the fingerprint is out of date.

**Comment by victor7376** [comment]
Well I put all the Spotify creds into this and it wasn't working even after I updated the fingerprint.  I'm pretty sure I know why its not working.

1, the code  for this clock doesn't use the HTTPS 443 port which when looking at the other Spotify displays for ESP8266 & ESP32 they DO.  they also include Spotify fingerprint (esp8266) AND Certificate (esp32) which are needed to make a https connection if none are provided then the connection is automatically closed.

I also noticed that the URL to the developer Spotify was incorrect, I did change this, but because the code doesn't use the secure libs it cant make the connection.

The only way I can see if this code can work is adding the Wi-Fi secure library into the code and making it use the 443 https connection.  Otherwise you'd need to use one of the other clocks I've just seen that are up to date and use the features.

Hope this helps you @drdudo00

**Comment by ElToberino** [comment]
Did you notice that **Spotify only works on ESP32**. 
Indeed a https connection must be established - in principle the ESP8266 could do this, but with all the other program features running, there's not enough heap left for the https connection and the ESP8266 crashes. That's the reason why it only works with an ESP32.

I justed tested the whole code on my ESP32 device and everything works fine. Spotify authentication process and API call are both running as intended. So I'm sure the problem isn't code related.

PS: 
Indeed, there seems to be a new fingerprint for Spotify - but the one I wrote into the code still seems to be valid, beacuse everything works fine for me. I will update the fingerprint in the next update.
But please note: The fingerprint is not individual or related to your account.
The link was just a hint, where you can get an updated fingerprint, if the current one is outdated.

**Comment by victor7376** [comment]
I've just got another clock running on an ESP8266 with https working with another api that requires https connection and fingerprint.  I have tested with an esp32 with just basic connection etc and that requires a cert coded into the esp32.

I'll have another try sometimes to see if I can get spotify to work with the esp32.

PS:
The fingerprint for spotify is now:
b9 79 6b ce fd 61 21 97 a7 02 90 ee da cd f0 a0 44 13 0e eb

Just went to spotify on chrome and clicked on the padlock - certififate valid - then the details tab - but I'm sure you already know this lol.  Just saves going to another site to get the new fingerprint.

**Comment by ElToberino** [comment]
With V1.3 I switched to secure connection to Spotify using a certificate.
The connection via fingerprint didn't work any longer with the new ESP32 core V1.0.5. 

I think a secure connection using fingerprint is well-implemented in the ESP8266 core - but as I wrote above: a secure connection with an ESP8266 doesn't work **with my program**. The reason is just the lack of RAM - with all the things running simultaniously (matrix animations, webserver etc) there is not enough memory left for a secure connection and the device crashes on the connection attempt.
